src/commands/register.ts

```diff
@@ -109,6 +115,38 @@ export default class ComponentRegister extends BaseCommand {
     }
   }
 
+  // eslint-disable-next-line max-params
+  private async uploadVolume(component_path: string, component_name: string, service_name: string, volume_name: string, volume: VolumeSpec, account: Account): Promise<VolumeSpec> {
+    const oras_plugin = await PluginManager.getPlugin<OrasPlugin>(this.app.config.getPluginDirectory(), OrasPlugin);
+
+    if (!volume.host_path) {
+      return classToClass(volume);
+    }
+
+    const base_folder = path.join(os.tmpdir(), `/${account.name}`);
+    fs.mkdirpSync(base_folder);
+    const file_name = `volume--${component_name}-${service_name}-${volume_name}`;
+    const file_path = `${account.name}/${file_name}`;
+    const registry_url = new URL(`/${file_path}:latest`, 'http://' + this.app.config.registry_host);
+
+    const updated_volume = classToClass(volume);
+    const host_path = path.resolve(component_path, untildify(updated_volume.host_path as string));
+    updated_volume.host_path = registry_url.href.replace('http://', '');
+
+    const output_file_location = path.join(os.tmpdir(), `${file_path}.tar`);
```

Comment:
Do we cleanup the file?

Reply:
This is a good pkg for handling tmp dir/files and it handles cleanup automatically
https://www.npmjs.com/package/tmp